### DISCUSSION TITLE
Overlay EXIF Data in Decimal not Fraction Format

### DISCUSSION THREAD

**Discussion by sparky5106** [ImageMagick/ImageMagick#2300] [Help] [answered]
After extensive searching and being new to Imagemagick and command line, I am trying to overlay EXIF data and have been successful in DateTimeOriginal and other fields. However, when overlaying GPSAltitude, it is formatted as 235/100 instead of 2.35. Is it possible to reduce the fraction to decimal and how? I thought maybe through an [fx:] but haven't found any resources.

Due to confidentiality of GPS positioning I am unable to share an image to give example of the file. But have uploaded a screenshot to show what I am talking about when being overlayed.

**Comment by fmw42** [comment]
What version of IM and what platform? On Unix, there are many tools to script such changes.

**Reply by sparky5106** [reply]
Apologies, I am using IM version 7.0.10-23 on Windows.

**Comment by snibgo** [comment]
I suggest using exiftool, so you get lines like:
```
GPS Altitude                    : 314.2 m Above Sea Level
```

**Reply by sparky5106** [reply]
Exiftool does display the value correctly, but I can not overlay the value using Exiftool from my understanding.

**Comment by snibgo** [comment]
You can put the altitude in an environment variable ...
```
for /F "usebackq tokens=2 delims==" %A in (`exiftool -GPSAltitude -args myfile.tif`) do set ALT=%A
```
... then use %ALT% in an IM command.

**Reply by sparky5106** [reply]
Thanks for the reply and believe I am over my head to start. So under system environment variables I added what you suggested above. Would I replace the myfile.tif with myfile.jpg? This would limit it to only this file when I have 10k+ to apply this to?

Also when you say use the %ALT% in a command, i tried replacing `[EXIF:GPSAltitude]` with `-annotate +45+45 "%[EXIF:DateTimeOriginal]\n[EXIF:GPSAltitude%ALT%]"` `-annotate +45+45 "%[EXIF:DateTimeOriginal]\n[EXIF:%ALT%]"` and `-annotate +45+45 "%[EXIF:DateTimeOriginal]\n%ALT%"`  how do you apply that to GPSAltitude?

**Comment by snibgo** [comment, marked as answer]
The value of %ALT% will be something like:
```
314.2 m Above Sea Level
```
This is a string. It isn't the name of an EXIF tag. If you want it in your annotation, just use it, for example:
```
-annotate +45+45 "%[EXIF:DateTimeOriginal]\n%ALT%"
```

**Reply by sparky5106** [reply]
Thanks for your help.
